preview/Win7Msix/Win7MSIXInstaller/MsixRequest.cpp

```diff
@@ -168,11 +168,20 @@ HRESULT MsixRequest::ProcessAddRequest()
         HandlerInfo currentHandler = AddHandlers[currentHandlerName];
         AutoPtr<IPackageHandler> handler;
         RETURN_IF_FAILED(currentHandler.create(this, &handler));
-        RETURN_IF_FAILED(handler->ExecuteForAddRequest());
+        if (!GetIsCancelClicked())

```

Comment:
>GetIsCancelClicked [](start = 13, length = 18)

Shouldn't this be checked in conjunction with the show UI option? Also, I do not believe we should add any code to this loop. If we need to handle a remove, we should do that explicitly by creating a remove request, or optionally in the array structure above, we can go from one statehandler to another in cancel case, and that can be the cancel handler.

Reply:
Modified the code as per design discussion to have an error handler which will remove the package in case of other handlers throw an error.